File: src/Python/622.CircularQueue.py

class MyCircularQueue:

    def __init__(self, k):
        """
        Initialize your data structure here. Set the size of the queue to be k.
        :type k: int
        """
        self.k = k
        self.queue = []

    def enQueue(self, value):
        """
        Insert an element into the circular queue. Return true if the operation is successful.
        :type value: int
        :rtype: bool
        """
        if len(self.queue) >= self.k:
            return False
        self.queue.append(value)
        return True

    def deQueue(self):
        """
        Delete an element from the circular queue. Return true if the operation is successful.
        :rtype: bool
        """
        if len(self.queue) <= 0:
            return False
        self.queue.pop(0)
        return True

    def Front(self):
        """
        Get the front item from the queue.
        :rtype: int
        """
        if len(self.queue) <= 0:
            return -1
        return self.queue[0]

    def Rear(self):
        """
        Get the last item from the queue.
        :rtype: int
        """
        if len(self.queue) <= 0:
            return -1
        return self.queue[-1]

    def isEmpty(self):
        """
        Checks whether the circular queue is empty or not.
        :rtype: bool
        """
        return len(self.queue) <= 0
        
    def isFull(self):
        """
        Checks whether the circular queue is full or not.
        :rtype: bool
        """
        return len(self.queue) >= self.k

if __name__ == '__main__':
    circularQueue = MyCircularQueue(3)
    print(circularQueue.enQueue(1))
    print(circularQueue.enQueue(2))
    print(circularQueue.enQueue(3))
    print(circularQueue.enQueue(4))
    print(circularQueue.Rear())
    print(circularQueue.isFull())
    print(circularQueue.deQueue())
    print(circularQueue.enQueue(4))
    print(circularQueue.Rear())
else:
    pass

File: src/Java/622.CircularQueue.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.HashMap;
import java.util.HashSet;

class MyCircularQueue {

    private LinkedList<Integer> list;
    private int size;

    /** Initialize your data structure here. Set the size of the queue to be k. */
    public MyCircularQueue(int k) {
        size = k;
        list = new LinkedList<>();
    }
    
    /** Insert an element into the circular queue. Return true if the operation is successful. */
    public boolean enQueue(int value) {
        if (list.size() >= size) 
            return false;
        list.add(value);
        return true;
    }
    
    /** Delete an element from the circular queue. Return true if the operation is successful. */
    public boolean deQueue() {
        if (list.size() <= 0) 
            return false;
        list.removeFirst();
        return true;
    }
    
    /** Get the front item from the queue. */
    public int Front() {
        if (list.size() <= 0) 
            return -1;
        return list.getFirst();
    }
    
    /** Get the last item from the queue. */
    public int Rear() {
        if (list.size() <= 0) 
            return -1;
        return list.getLast();
    }
    
    /** Checks whether the circular queue is empty or not. */
    public boolean isEmpty() {
        return list.size() <= 0;
    }
    
    /** Checks whether the circular queue is full or not. */
    public boolean isFull() {
        return list.size() >= size;
    }
}

